ts/index.ts

```diff
@@ -352,11 +352,11 @@ class Gossipsub extends BasicPubsub {
     const prune = this._handleGraft(id, controlMsg.graft)
     this._handlePrune(id, controlMsg.prune)
 
-    if (!iwant || !ihave || !prune) {
+    if (!iwant && !ihave && !prune) {
       return
     }
 
-    const outRpc = createGossipRpc(ihave, { iwant: [iwant], prune })
+    const outRpc = createGossipRpc(ihave, { iwant: iwant ? [iwant] : [], prune })
```

Comment:
```suggestion
    const outRpc = createGossipRpc(ihave, { iwant: iwant || [], prune })
```

perhaps we can simplify?